### PR TITLE
added StatsRoute serving metrics over http

### PR DESCRIPTION
**Pull request by khodzha**

### CODE REVIEW

File: src/app/mod.rs [hunk 1]
```diff
@@ -380,13 +387,37 @@ async fn handle_message(
...
+                                        .get("metric")
+                                        .and_then(|k| k.as_str())
+                                        .and_then(|s| s.rsplit('.').nth(0));
+                                    let value = metric.get("value").and_then(|v| v.as_i64());
+                                    match (key, value) {
+                                        (Some(k), Some(v)) => {
+                                            stats_route.update((sender.clone(), k.to_owned()), v);
+                                        }
+                                        _ => {}
+                                    }
```
Comment (feymartynov) [inline src/app/mod.rs:414]: if let

File: src/app/stats_route.rs [hunk 2]
```diff
@@ -0,0 +1,99 @@
...
+                    StatsRouteCommand::GetStats(chan) => {
+                        if let Err(e) = chan.send(Self::collect_hashmap(&s)) {
+                            error!("Error sending get_stats resp: {:?}", e);
+                        }
+                    }
+                };
+            }
+        });
+
+        let my_app = Self { tx };
```
Comment (feymartynov) [inline src/app/stats_route.rs:44]: my_app?

File: src/app/mod.rs [hunk 3]
```diff
@@ -380,13 +387,37 @@ async fn handle_message(
...
-                    json_payload
+                let json_payload = match event.properties().label() {
+                    Some("metric.pull") => {
+                        let metrics = metrics::get_metrics(qc_handle, json_payload)?;
+                        serde_json::to_value(metrics)
+                            .context("Failed to serialize message payload")?
+                    }
+                    Some("metric.create") => {
+                        if let Some(stats_route) = stats_route {
+                            if json_payload.is_array() {
```
Comment (feymartynov) [inline src/app/mod.rs:402]: It would be better do deserialize the payload with serde rather than accessing it through `get`.

File: Cargo.toml [hunk 4]
```diff
@@ -21,3 +21,5 @@ svc-agent = { version = "0.14", features = ["queue-counter"] }
 svc-authn = { version = "0.6", features = ["jose"] }
 svc-error = { version = "0.1", features = ["svc-agent", "svc-authn", "sentry-extension"] }
 woothee = "0.11.0"
+tower-web = "0.3"
```
Comment (manifest) [inline Cargo.toml:24]: TowerWeb is dead. I assume Tide would be a better choice.

File: src/app/mod.rs [hunk 5]
```diff
@@ -380,13 +387,37 @@ async fn handle_message(
...
+                                for metric in json_payload.as_array().unwrap() {
+                                    let key = metric
+                                        .get("metric")
+                                        .and_then(|k| k.as_str())
+                                        .and_then(|s| s.rsplit('.').nth(0));
```
Comment (manifest) [inline src/app/mod.rs:407]: Let's change our convention to be Prometheus compatible instead.

File: src/app/mod.rs (hunk 3 above)
Comment (manifest) [reply src/app/mod.rs:402]: [Everything sent to Telemetry topic](https://github.com/netology-group/telemetry/blob/024d418c981dcb258111bb7a029663273235e868/src/app/mod.rs) must have the same format.

File: Cargo.toml (hunk 4 above)
Comment (khodzha) [reply Cargo.toml]: replaced with tide

File: src/app/mod.rs (hunk 3 above)
Comment (khodzha) [reply src/app/mod.rs]: added deserialization via trait, addressed formats discrepancies

**Review by manifest — APPROVED** [review]

**Review by feymartynov — APPROVED** [review]

**Pull request merged**